pkg/collector/reconcile/config_replace.go

```diff
@@ -71,13 +87,16 @@ func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
 		}
 	}
 
-	updPromCfgMap := make(map[string]interface{})
-	if err := mapstructure.Decode(cfg, &updPromCfgMap); err != nil {
-		return "", err
+	if EnableTargetAllocatorRewrite.IsEnabled() {
```

Comment:
I wonder if we could correlate this with the flag enabling TA, to prevent misconfiguration? We could log a warning for example if we discover TA is not enabled but user tried to enable allocator rewrite.

Reply:
This will only work if the TA is enabled the top of the function makes a check
```
func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
	if !instance.Spec.TargetAllocator.Enabled {
		return instance.Spec.Config, nil
	}
```

Reply:
Yes, so if user _disables_ TA but _enables_ allocator rewrite, it will not have any effect but user won't know what's wrong. Not super vital, but nice to have.